File: backend/UniUti/models/Monitoria.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using UniUti.models.Enum;
using UniUti.Models.Base;

namespace UniUti.Models
{
    [Table("monitorias")]
    public class Monitoria : EntidadeBase
    {
        [Required]
        [Column("solicitante")]
        public virtual Usuario? Solicitante { get; set; }

        [Column("prestador")]
        public virtual Usuario? Prestador { get; set; }

        [Required]
        [Column("descricao")]
        [StringLength(500)]
        public string? Descricao { get; set; }

        [Required]
        [Column("disciplina")]
        [StringLength(500)]
        public virtual Disciplina? Disciplina { get; set; }

        [Required]
        [Column("data_solicitacao")]
        public DateTime? DataCriacao { get; set; }

        [Required]
        [Column("status_solicitacao")]
        public StatusSolicitacao? StatusSolicitacaco { get; set; }
    }
}

File: app/lib/monitoria/domain/monitoria.dart
import '../../aluno/domain/aluno.dart';
import '../../disciplina/domain/disciplina.dart';
import 'status.dart';

class Monitoria {
  int id;
  String titulo;
  String descricao;
  Disciplina disciplina;
  Aluno? prestador;
  Aluno? solicitante;
  Status status;
  List<Aluno> pendencias;
  Monitoria({
    required this.id,
    required this.titulo,
    required this.descricao,
    required this.disciplina,
    this.prestador,
    this.solicitante,
    required this.status,
    required this.pendencias,
  }) {
    assert((prestador != null && solicitante != null) ||
        (prestador == null && solicitante != null) ||
        (prestador != null && solicitante == null));
  }
}
